src/JsonSchema/SchemaFactory.php

```diff
@@ -109,7 +109,7 @@ public function buildSchema(string $className, string $format = 'json', string $
             return $schema;
         }
 
-        $definition = new \ArrayObject(['type' => 'object', 'additionalProperties' => $serializerContext[AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES] ?? true]);
+        $definition = new \ArrayObject(['type' => 'object', 'additionalProperties' => $serializerContext[AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES] ?? false]);
```

Comment:
I don't think this is the right fix. Allowing extra attributes should be `true` by default. We just want to remove the `additionalProperties=true` to the definition.

Reply:
Actually, do we really want to explicitly allow extra properties? It's technically allowed because of a limitation of the Symfony serializer (that has now been fixed, and we may disallow extra properties by default in API Platform 3), but it's more a side effects than something really intended isn't it? It looks good to me to not set this flag unless extra properties are explicitly allowed by the user.

Reply:
You mean not setting `additionalProperties` at all for the definition when allow_extra_attributes is not set in the $serializerContext?

Reply:
Yes! WDYT @alanpoulain @soyuka?

Reply:
OK so to be clear, if there is no `additionalProperties` in the definition, additional properties are **allowed** by default (see https://json-schema.org/understanding-json-schema/reference/object.html).
So I don't think we need to set it explicitly if `allow_extra_attributes` is set to `true`.
If there is no `allow_extra_attributes` in the serializer context, its value is considered `true` by default. So I think we should keep it this way for JSON Schema and don't explicitly set it too.
We *only* need to set it explicitly to `false` when `allow_extra_attributes` is set to `false`.
This is my opinion :slightly_smiling_face:

Reply:
I don't agree. It's not excepted to add extra properties by default to most API Platform documents and the additional property displayed in Swagger UI pollutes the example, degrading the user experience when using the sandbox (or just when reading the docs of the API).

Reply:
Si my proposal is to not set the key only when the context option is explicitly set to `true`.

Reply:
I don't follow you. The additional property is displayed in Swagger UI *because* `additionalProperties` is explicitly set to true.
If you follow my proposal, this problem doesn't exist anymore.

Reply:
Alan's proposal looks good to me

Reply:
Ok, got it! Let's do this.